skorch/tests/test_net.py

```diff
@@ -1593,6 +1593,74 @@ def test_fit_with_dataset_stratified_without_explicit_y_raises(
         msg = "Stratified CV requires explicitely passing a suitable y."
         assert exc.value.args[0] == msg
 
+    @pytest.fixture
+    def dataset_1_item(self):
+        class Dataset(torch.utils.data.Dataset):
+            def __len__(self):
+                return 100
+
+            def __getitem__(self, i):
+                return 0.0
+        return Dataset
+
+    def test_fit_with_dataset_one_item_error(
+            self, net_cls, module_cls, dataset_1_item):
+        net = net_cls(module_cls, train_split=None)
+        with pytest.raises(ValueError) as exc:
+            net.fit(dataset_1_item(), None)
+
+        msg = ("You are using a non-skorch dataset that returns 1 value. "
+               "Remember that for skorch, Dataset.__getitem__ must return "
+               "exactly 2 values, X and y (more info: "
+               "https://skorch.readthedocs.io/en/stable/user/dataset.html).")
```

Comment:
Just noticed this: can't you omit duplicating this message here as well and simply reference the corresponding variable?
```python
msg = skorch.dataset.ERROR_MSG_MORE_THAN_2_ITEMS
```

Reply:
I'm a little bit torn about this kind of thing, because it makes the test look circular (the message can never be wrong when it is just referenced). But I do also see the advantages, so not sure overall.